### PR TITLE
add documentation to several Hash methods

### PR DESCRIPTION
**Pull request by adlerhsieh**

### CODE REVIEW

File: src/hash.cr [hunk 1]
```diff
@@ -613,6 +619,11 @@ class Hash(K, V)
     self
   end
 
+  # Returns a `Hash`, which contains each element in *ary1* and *ary2* as key and value.
```
Comment (ysbaddaden) [inline src/hash.cr]: What about:

> Zips two arrays into a hash, taking keys from *ary1* and values from *ary2*.

File: src/hash.cr [hunk 2]
```diff
@@ -436,6 +436,12 @@ class Hash(K, V)
     keys
   end
 
+  # Returns a new `Array` with all the values.
```
Comment (ysbaddaden) [inline src/hash.cr]: What about:

> Returns only the values as an `Array`.

File: src/hash.cr [hunk 3]
```diff
@@ -654,14 +665,47 @@ class Hash(K, V)
     @first.try &.value
   end
 
+  # Deletes and returns the first key-value pair in the hash,
+  # or raises `IndexError` if the hash does not have any.
```
Comment (ysbaddaden) [inline src/hash.cr]: > if the hash is empty.

**Comment by omninonsense** [comment]
This would be twice as cool with some examples :grin: (but keep them basic)!

**Comment by ysbaddaden** [comment]
@omninonsense well, there _are_ examples

**Comment by omninonsense** [comment]
@ysbaddaden Oh you're right! I think it was collapsed on my phone earlier. So, it already *is* twice as cool.

**Comment by adlerhsieh** [comment]
Thanks for suggestions. Updated.

File: src/hash.cr [hunk 4]
```diff
@@ -654,14 +665,47 @@ class Hash(K, V)
     @first.try &.value
   end
 
+  # Deletes and returns the first key-value pair in the hash,
+  # or raises `IndexError` if the hash is empty.
+  #
+  # ```
+  # hash = {"foo" => "bar", "baz" => "qux"}
+  # hash.shift
```
Comment (jhass) [inline src/hash.cr]: Maybe annotate the returned value here?

File: src/hash.cr [hunk 5]
```diff
@@ -654,14 +665,47 @@ class Hash(K, V)
...
+  # 
+  # hash = {} of String => String
+  # hash.shift # => Index out of bounds (IndexError)
+  # ```
   def shift?
     shift { nil }
   end
 
+  # Deletes and returns the first key-value pair in the hash.
+  # Yields the given block if the hash is empty.
```
Comment (jhass) [inline src/hash.cr]: `Yields to`, yielding is the operation happening on the block, not another operation that the block is passed to.

File: src/hash.cr [hunk 6]
```diff
@@ -654,14 +665,47 @@ class Hash(K, V)
...
+  #
+  # ```
+  # hash = {"foo" => "bar", "baz" => "qux"}
+  # hash.shift { true } # => {"foo" => "bar"}
```
Comment (jhass) [inline src/hash.cr]: This is not what's returned.

File: src/hash.cr [hunk 7]
```diff
@@ -698,6 +742,12 @@ class Hash(K, V)
     true
   end
 
+  # Generates an `Int` hash value based on its content.
```
Comment (jhass) [inline src/hash.cr]: `See \`Object#hash\`` ?

**Comment by adlerhsieh** [comment]
Thanks for correction. Updated.

**Pull request merged**